TestInstance create_overlay() extra_configs with per-platform filtering broken

**Describe the bug**
While expanding the test coverage for twister, I have noticed that platform-based filtering for extra configs in `testinstance.py`'s `create_overlay()` behaves in an unexpected manner. Instead of recognizing the colon-separated `platform` prefix as an indication to check for the platform, the whole config string is treated as a config.

**Expected behavior**
`TestInstance.create_overlay()` platform filtering for extra_configs to properly filter them.

**Impact**
TestInstance platform filtering for extra configs is broken completely.

**Logs and console output**
```
============================================================== FAILURES ===============================================================
___________________ test_create_overlay[False-False-False-coverage_platform11-native-extra_configs11-CONFIG_NULL=y] ___________________

class_testplan = <twisterlib.testplan.TestPlan object at 0x7fc364d51f00>
all_testsuites_dict = {'scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app': <twisterlib.testsuite.TestSuite object...ster/test_data/testsuites/tests/test_b/test_b.check_1': <twisterlib.testsuite.TestSuite object at 0x7fc364e292a0>, ...}
platforms_list = [<demo_board_2 on x86_demo>, <demo_board_3 on arm_demo>, <demo_board_1 on arm_demo>], enable_asan = False
enable_ubsan = False, enable_coverage = False, coverage_platform = ['demo_board_2'], platform_type = 'native'
extra_configs = ['platform:demo_board_2:CONFIG_NULL=y'], expected_content = 'CONFIG_NULL=y'

    @pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content", TESTDATA_2)
    def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content):
        """Test correct content is written to testcase_extra.conf based on if conditions."""
        class_testplan.testsuites = all_testsuites_dict
        testcase = class_testplan.testsuites.get('scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app')

        if extra_configs:
            testcase.extra_configs = extra_configs

        class_testplan.platforms = platforms_list
        platform = class_testplan.get_platform("demo_board_2")

        testinstance = TestInstance(testcase, platform, class_testplan.env.outdir)
        platform.type = platform_type
>       assert testinstance.create_overlay(platform, enable_asan, enable_ubsan, enable_coverage, coverage_platform) == expected_content
E       AssertionError: assert 'platform:dem...CONFIG_NULL=y' == 'CONFIG_NULL=y'
E         - CONFIG_NULL=y
E         + platform:demo_board_2:CONFIG_NULL=y

tests/twister/test_testinstance.py:93: AssertionError
```

**Environment (please complete the following information):**
 - OS: Linux
 - N/A
 - Commit SHA: `28a18ece433ab1534392896acd7bb28bb3dd05a2`

--- a/scripts/pylib/twister/twisterlib/testinstance.py
+++ b/scripts/pylib/twister/twisterlib/testinstance.py
@@ -227,7 +227,7 @@ class TestInstance:
                 if cond_config[0] == "arch" and len(cond_config) == 3:
                     if self.platform.arch == cond_config[1]:
                         new_config_list.append(cond_config[2])
-                elif cond_config[0] == "plaform" and len(cond_config) == 3:
+                elif cond_config[0] == "platform" and len(cond_config) == 3:
                     if self.platform.name == cond_config[1]:
                         new_config_list.append(cond_config[2])
                 else:

--- a/scripts/tests/twister/test_testinstance.py
+++ b/scripts/tests/twister/test_testinstance.py
@@ -56,22 +56,30 @@ def test_check_build_or_run(class_testplan, monkeypatch, all_testsuites_dict, pl
     assert not run
 
 TESTDATA_2 = [
-    (True, True, True, ["demo_board_2"], "native", '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y\nCONFIG_UBSAN=y'),
-    (True, False, True, ["demo_board_2"], "native", '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y'),
-    (False, False, True, ["demo_board_2"], 'native', '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
-    (True, False, True, ["demo_board_2"], 'mcu', '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
-    (False, False, False, ["demo_board_2"], 'native', ''),
-    (False, False, True, ['demo_board_1'], 'native', ''),
-    (True, False, False, ["demo_board_2"], 'native', '\nCONFIG_ASAN=y'),
-    (False, True, False, ["demo_board_2"], 'native', '\nCONFIG_UBSAN=y'),
+    (True, True, True, ["demo_board_2"], "native", None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y\nCONFIG_UBSAN=y'),
+    (True, False, True, ["demo_board_2"], "native", None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y'),
+    (False, False, True, ["demo_board_2"], 'native', None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
+    (True, False, True, ["demo_board_2"], 'mcu', None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
+    (False, False, False, ["demo_board_2"], 'native', None, ''),
+    (False, False, True, ['demo_board_1'], 'native', None, ''),
+    (True, False, False, ["demo_board_2"], 'native', None, '\nCONFIG_ASAN=y'),
+    (False, True, False, ["demo_board_2"], 'native', None, '\nCONFIG_UBSAN=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["CONFIG_LOG=y"], 'CONFIG_LOG=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["arch:x86_demo:CONFIG_LOG=y"], 'CONFIG_LOG=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["arch:arm_demo:CONFIG_LOG=y"], ''),
+    (False, False, False, ["demo_board_2"], 'native', ["platform:demo_board_2:CONFIG_LOG=y"], 'CONFIG_LOG=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["platform:demo_board_1:CONFIG_LOG=y"], ''),
 ]
 
-@pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, expected_content", TESTDATA_2)
-def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, expected_content):
-    """Test correct content is written to testcase_extra.conf based on if conditions
-    TO DO: Add extra_configs to the input list"""
+@pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content", TESTDATA_2)
+def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content):
+    """Test correct content is written to testcase_extra.conf based on if conditions."""
     class_testplan.testsuites = all_testsuites_dict
     testcase = class_testplan.testsuites.get('scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app')
+
+    if extra_configs:
+        testcase.extra_configs = extra_configs
+
     class_testplan.platforms = platforms_list
     platform = class_testplan.get_platform("demo_board_2")
 

--- a/subsys/testsuite/ztest/src/ztress.c
+++ b/subsys/testsuite/ztest/src/ztress.c
@@ -461,9 +461,9 @@ static int ztress_cpu_clock_to_sys_clock_check(void)
 	}
 
 	t = sys_clock_tick_get_32() - t;
-	/* Threshold is arbitrary. Derived from nRF platorm where CPU runs at 64MHz and
-	 * system clock at 32kHz (sys clock interrupt every 1950 cycles). That ratio is
-	 * ok even for no optimization case.
+	/* Threshold is arbitrary. Derived from nRF platform where CPU runs
+	 * at 64MHz and system clock at 32kHz (sys clock interrupt every 1950
+	 * cycles). That ratio is ok even for no optimization case.
 	 * If some valid platforms are cut because of that, it can be changed.
 	 */
 	cpu_sys_clock_ok = t <= 12;

--- a/tests/kernel/mem_heap/shared_multi_heap/src/main.c
+++ b/tests/kernel/mem_heap/shared_multi_heap/src/main.c
@@ -120,8 +120,8 @@ static void fill_multi_heap(void)
 		 * address retrieved by DT at run-time because the SMH
 		 * framework expects virtual addresses.
 		 *
-		 * For MPU-enabled plaform the code is assuming that the region
-		 * are configured at build-time, so no map is needed.
+		 * For MPU-enabled platform the code is assuming that the
+		 * region are configured at build-time, so no map is needed.
 		 */
 		smh_reg_map(&reg_map->region);
 #endif /* CONFIG_MMU */

--- a/tests/lib/devicetree/api/src/main.c
+++ b/tests/lib/devicetree/api/src/main.c
@@ -1369,7 +1369,7 @@ ZTEST(devicetree_api, test_arrays)
 ZTEST(devicetree_api, test_foreach)
 {
 	/*
-	 * We don't know what plaform we are running on, so settle for
+	 * We don't know what platform we are running on, so settle for
 	 * some basic checks related to nodes we know are in our overlay.
 	 */
 #define IS_ALIASES(node_id) + DT_SAME_NODE(DT_PATH(aliases), node_id)

